localstack-core/localstack/aws/api/firehose/__init__.py

```diff
@@ -1353,6 +1488,7 @@ def create_delivery_stream(
         msk_source_configuration: MSKSourceConfiguration = None,
         snowflake_destination_configuration: SnowflakeDestinationConfiguration = None,
         iceberg_destination_configuration: IcebergDestinationConfiguration = None,
+        database_source_configuration: DatabaseSourceConfiguration = None,
```

Comment:
@bentsku that's what you meant with the kwargs not helping in these cases because this will be caught by the unit tests, right?

Reply:
The kwargs would fix the problem, but we still want to be somewhat strict and keep the signatures in sync (which is why we have the unit test).